load-tests/load-tester/src/main/java/io/camunda/zeebe/ProcessInstanceStartMeter.java

```diff
@@ -77,35 +77,24 @@ private void checkForProcessInstances() {
     }
 
     LOG.debug("Current instances awaiting {}", startedInstances.size());
-    final List<Long> list = startedInstances.stream().map(k -> k.processInstanceKey).toList();
     availabilityChecker
-        .findAvailableInstances(list)
+        .findAvailableInstances(List.copyOf(startedInstances.keySet()))
         .whenCompleteAsync(
             (availableInstances, error) -> {
               if (error != null) {
                 LOG.error("Error while checking for available process instances", error);
                 return;
               }
 
-              checkAvailableInstances(availableInstances);
+              LOG.debug("Available process instances items: {} ", availableInstances.size());
+              availableInstances.stream()
+                  .map(startedInstances::get)
+                  .filter(Objects::nonNull)
+                  .forEach(this::recordInstanceAvailable);
```

Comment:
🤔 I guess it is better to simply use one `for loop` to get the record and pass it into the method. I think with the stream we have now again two loops.